durable-storage/src/database.rs

```diff
@@ -118,3 +121,46 @@ impl Database {
         value.ok_or(DatabaseError::KeyNotFound)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::path::Path;
+    use std::sync::Arc;
+
+    use tokio::runtime::Handle;
+
+    use super::Database;
+    use crate::merkle_layer::Key;
+    use crate::merkle_worker::MerkleWorker;
+    use crate::persistence_layer::PersistenceLayer;
+    use crate::repo::DirectoryManager;
+
+    fn new_database(handle: &Handle) -> Database {
+        let directory_manager =
+            DirectoryManager::new(Path::new("/tmp")).expect("Should be a valid directory");
+
+        let persistent: Arc<PersistenceLayer> = PersistenceLayer::new(&directory_manager)
+            .expect("Creating a persistence layer should succeed")
+            .into();
+
+        let merkle = MerkleWorker::new(handle, persistent.clone())
+            .expect("Creating a Merkle worker should succeed");
+
+        Database { persistent, merkle }
+    }
+
+    #[test]
+    fn test_database_write() {
+        let runtime = tokio::runtime::Builder::new_multi_thread()
+            .worker_threads(2)
+            .build()
+            .expect("Creating a Tokio runtime should succeed");
+        let handle = runtime.handle();
+        let mut database = new_database(handle);
+
+        let key = Key::new(&[0]).expect("Size less than KEY_MAX_SIZE");
+        let data = "write";
+
+        database.write(&key, 0, data.as_ref()).unwrap();
```

Comment:
```suggestion
        database.write(&key, 0, data.as_ref()).expect("Writing should succeed");
```

Everything else is using expect in this test, it feels strange for the thing actually under test to use `unwrap`.

(though in all honesty I wouldn't mind it if `.unwrap()` was used everywhere else in `test_database_write` - as they are all expected to succeed)